src/aks-preview/azext_aks_preview/managed_cluster_decorator.py

```diff
@@ -2971,6 +3046,45 @@ def update_sku(self, mc: ManagedCluster) -> ManagedCluster:
             )
         return mc
 
+    def update_upgrade_settings(self, mc: ManagedCluster) -> ManagedCluster:
+        """Update upgrade settings for the ManagedCluster object.
+        :return: the ManagedCluster object
+        """
+        self._ensure_mc(mc)
+
+        # There is a limitation on differentiating empty list vs. not set in update requests.
+        # In such case, we'll use a workaround here to disable it by setting the until field to the current time, to make the overrides no longer effective.
+        # For now there's only one allowed override so we can return early here.
+        if self.context.get_disable_upgrade_ignore_kubernetes_deprecations() == True:
+            if mc.upgrade_settings.override_settings.control_plane_overrides == [CONST_IGNORE_KUBERNETES_DEPRECATIONS]:
+                if mc.upgrade_settings.override_settings.until is None or parse(mc.upgrade_settings.override_settings.until).timestamp() > datetime.datetime.utcnow().timestamp():
```

Comment:
Though the value of this time is returned by rp (means it should be valid), I'm still a little concerned, maybe catch possible exceptions when parsing the string would be better

Reply:
Updated.